client/my-sites/checkout/src/test/credit-card.tsx

```diff
@@ -143,4 +144,35 @@ describe( 'Credit card payment method', () => {
 			expect( processorFunction ).not.toHaveBeenCalled();
 		} );
 	} );
+
+	it( 'displays error message overlay when a credit card field is empty and submit is clicked', async () => {
+		const user = userEvent.setup();
+		const store = createCreditCardPaymentMethodStore( {} );
+		const paymentMethod = getPaymentMethod( store );
+		const processorFunction = jest.fn( () => Promise.resolve( makeSuccessResponse( {} ) ) );
+		render(
+			<TestWrapper
+				paymentMethods={ [ paymentMethod ] }
+				paymentProcessors={ { card: processorFunction } }
+			></TestWrapper>
+		);
+
+		await waitFor( () => expect( screen.getByText( activePayButtonText ) ).not.toBeDisabled() );
+
+		// Partially fill the form, leaving security code field empty
+		await user.type( screen.getAllByLabelText( /Cardholder name/i )[ 1 ], customerName );
+		await user.type( screen.getByLabelText( /Card number/i ), cardNumber );
+		await user.type( screen.getByLabelText( /Expiry date/i ), cardExpiry );
+		//await user.type( screen.getAllByLabelText( /Security code/i )[ 0 ], '' );
+
+		// Try to submit the form
+		await user.click( await screen.findByText( activePayButtonText ) );
+
+		// Verify the error message overlay appears
+		await waitFor( () => {
+			expect( screen.getByText( /Something seems to be missing/i ) );
```

Comment:
This test seems to be failing.

Looks like if you run `it.only` for the last test, and remove the debug statement, the test passes. It is possible there is a leaky test above where some sort of async/promise is resolving after the calling test finishes and pollutes the scope for the subsequent test(s). Try running `it.skip` on the previous tests one by one to see if the last test passes at some point

Reply:
Thank you, I have modified the test with some jest mocking so that the test passes.